### PR TITLE
2021 12 data tables

### PR DESCRIPTION
**Pull request by xwatkins**
## Purpose
Use the new `protvista-datatable` component

## Approach
Refactor use of `protvista-datatable`, where possible simplify the code. In the process, discovered some issues and reported them on Jira:
https://www.ebi.ac.uk/panda/jira/browse/TRM-26691
https://www.ebi.ac.uk/panda/jira/browse/TRM-26690
https://www.ebi.ac.uk/panda/jira/browse/TRM-26692
https://www.ebi.ac.uk/panda/jira/browse/TRM-26693

## Testing
Update snapshots, which now include table contents.

## Checklist
- [ ] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

**Review by dlrice — COMMENTED** [review]
Should interaction work at this point:

1. Changing the start/end location in the zoom track filters the data table rows
2. Clicking a data table row highlights the corresponding item in the track 

I tried in P05067 > Function > Features but couldn't see the interaction happening.

File: package.json [hunk 1]
```diff
@@ -109,26 +109,26 @@
     "deep-freeze": "0.0.1",
     "franklin-sites": "0.0.163",
     "history": "4.10.1",
-    "idb": "6.1.4",
-    "interaction-viewer": "3.1.2",
+    "idb": "6.1.3",
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply package.json]: Looks like it was a git merge issue! Back to 6.1.4 now.

**Comment by xwatkins** [comment]
All should now be addressed.

**Review by dlrice — APPROVED** [review]
Looks good to me.

A few issues which I think go back to the nightingale's data table:

Clicking on the Sequence in the row contents updates start/end and so highlights in the track but does not change the row to yellow. Clicking on the "parent" row however does change the row to yellow.

Coming across a few errors in the console:

1. At http://localhost:8080/uniprotkb/P05067/entry
```
TypeError: Failed to execute 'observe' on 'MutationObserver': parameter 1 is not of type 'Node'.
```
and points at:
```
    this.mutationObserver.observe(this.querySelector("table tbody"), {
            characterData: true,
            childList: true,
            subtree: true,
        });
```

2. At http://localhost:8080/uniparc/UPI000000002B/feature-viewer

```
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'scrollIntoView')
```
and points at:
```
element.scrollIntoView({ behavior: "smooth", block: "center" });
```

**Comment by xwatkins** [comment]
Should be sorted now! also fixed a typing issue and removed lit-html

**Pull request merged**